src/consensus/pbft/libbyz/message.h

```diff
@@ -118,6 +119,36 @@ class Message
   const char* stag();
   // Effects: Returns a string with tag name
 
+  class MsgBufCounter
+  {
+  public:
+    MsgBufCounter(Message_rep* msg_, int max_size_, bool should_delete_) :
+      msg(msg_),
+      max_size(max_size_),
+      should_delete(should_delete_),
+      counter(0)
+    {}
+    ~MsgBufCounter()
+    {
+      if (should_delete)
+      {
+        free(msg);
+      }
+    }
+
+    Message_rep* msg; // Pointer to the contents of the message.
+    int max_size; // Maximum number of bytes that can be stored in "msg"
+                  // or "-1" if this instance is not responsible for
+                  // deallocating the storage in msg.
+    int32_t counter;
+    bool should_delete = false;
```

Comment:
I don't understand the ownership model of this object. We have a `should_delete` bool, but also a comment that `max_size == -1` may affect deallocation, and on top of that a reference-count wrapper.

Reply:
I will add this as a comment in the code:

There are 2 types of messages (in regards to memory ownership).
1 - the message owns its own buffer (This is the normal case)
2 - A projection on top of a message on an array

The comment about -1 is no longer correct and I am deleting it